lib/BC/SleighLifter.cpp

```diff
@@ -585,6 +595,21 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
   static std::unordered_set<OpCode> INTEGER_COMP_OPS;
 
 
+  LiftStatus TerminateBlockWithCondition(llvm::Value *condition) {
+    llvm::IRBuilder<> ir(this->target_block);
+    this->target_block = llvm::BasicBlock::Create(
+        this->context, "continuation", this->target_block->getParent());
+    ir.CreateCondBr(condition, this->exit_block, this->target_block);
+    return LiftStatus::kLiftedInstruction;
+  }
+
+  void TerminateBlock() {
```

Comment:
Name is kinda misleading - I would try to say that control flow is also redirected somewhere, because just `TerminateBlock` can also mean a simple `return` or other terminator is there.